src/vma/dev/ib_ctx_handler.cpp

```diff
@@ -123,22 +122,21 @@ ib_ctx_handler::ib_ctx_handler(struct ibv_context* ctx, ts_conversion_mode_t ctx
 	// Create pd for this device
 	m_p_ibv_pd = ibv_alloc_pd(m_p_ibv_context);
 	if (m_p_ibv_pd == NULL) {
-		ibch_logpanic("ibv device %p pd allocation failure (ibv context %p) (errno=%d %m)", 
+		ibch_logpanic("ibv device %p pd allocation failure (ibv context %p) (errno=%d %m)",
 			    m_p_ibv_device, m_p_ibv_context, errno);
 	}
-
-	memset(&m_ibv_device_attr, 0, sizeof(m_ibv_device_attr));
-	vma_ibv_device_attr_comp_mask(m_ibv_device_attr);
-	IF_VERBS_FAILURE(vma_ibv_query_device(m_p_ibv_context, &m_ibv_device_attr)) {
-		ibch_logerr("ibv_query_device failed on ibv device %p (ibv context %p) (errno=%d %m)", 
+	m_p_ibv_device_attr = new vma_ibv_device_attr();
```

Comment:
https://stackoverflow.com/questions/7546620/operator-new-initializes-memory-to-zero